src/Microsoft.DotNet.Wpf/src/System.Xaml/System.Xaml.csproj

```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <ProjectGuid>{9AC36357-34B7-40A1-95CA-FE9F46D089A7}</ProjectGuid>
     <AssemblyName>System.Xaml</AssemblyName>
-    <TargetFrameworks>netcoreapp3.0</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net461;netcoreapp2.0;netcoreapp2.1;netcoreapp3.0</TargetFrameworks>
     <Platforms>AnyCPU;x64</Platforms>
     
```

Comment:
Targeting net461 means you are trying to replace the System.Xaml which is inbox in .NETFramework.  This can be done, but would require that this assembly be versioned to 4.1.0.0 to break out of unification.  It'd also put this assembly into the "oob" state which we've been trying to avoid for a number of reasons on desktop (search for System.Net.Http in the corefx repository to get an idea).  /cc @karelz 

Targeting NETStandard2.0 and maintaining identical surface area to desktop with a placeholder in the package for desktop is another option.  This is what we do for things like Microsoft.CSharp and other assemblies.  The downside is you can only add API to "new" frameworks like NETCoreApp3.0 and netstandard2.1.

There are other options that involve renaming the assembly and using facades, but I think I'd want the WPF folks to weigh in on the direction here.

Reply:
This is a non-trivial decision (experts in area need to understand all implications which are fairly complicated). A bad decision can easily bite the community, the maintainers and everyone.
We should be extra careful with these things - see my comment https://github.com/dotnet/wpf/issues/46#issuecomment-444293310
I will mark it NO MERGE until we have clear understanding here on direction. I expect it may take days/weeks (I am around CoreFX out-of-band releases for quite some time and the key thing I learned is "be scared around them").

Reply:
One option would be something similar to what @ericstj suggested with a ref-assembly and placeholder package for netstandard2.0/netcoreapp2.0 and only adding the new APIs to netcoreapp3.0 and netstandard2.1 surfaces. That would satisfy the use cases of @wieslawsoltes as well as other groups currently using 3rd-party solutions for XAML parsing off-.NET Framework.

And then the WPF team could use the .NET Standard 2.1+ and .NET Core 3.0+ targets for new API surface area, which people could move up to as they desire.

Reply:
>ref-assembly and placeholder package for netstandard2.0

Xamarin frameworks doesn't have a proper version of System.Xaml at the moment, so it would be nice to have a proper version for netstandard2.0 build with placeholder for net461.

Reply:
> I don't get why you'd have to compile for both Netcoreapp 2.0, 2.1 and 3.0 as well as netstandard 2.0. Netstandard2.0 would cover all of these (and also as mentioned earlier I think it should be at least netstandard 2.1).

Because of issues with `ValueSerializerAttribute` type missing in those frameworks I had to add additional targets for compile and use `System.Xaml` in my tests.